### PR TITLE
Added configuration parameters for PoS contract (multi-sig keys and quorum)

### LINKED ISSUE #3625: Create pubKey configuration for POS multi-sig wallet
## Overview

Proof of Stake contract has hard-coded value for multi-sig vault used for slashed stake. Instead of hard-coded public keys and quorum size they should be supplied as configuration parameters when genesis block is created. 

Currently rnode use [pureconfig](https://pureconfig.github.io/docs/index.html) to read config.

Pureconfig support `HOCON`, json, java`.properties` format. They all support array config.

#### Example
HOCON
```
pubKeys = [
   pubKey1, pubKey2, pubKey3
]
```

Json
```
{"pubKeys":[pubKey1, pubKey2, pubKey3]}
```

[Jave .properties](https://help.mulesoft.com/s/question/0D52T00004mXUpxSAG/how-to-declare-array-in-properties-file)(Doesn't care actually, no one uses this)

### Design

We should introduce this config like [bond-maximum](https://github.com/rchain/rchain/blob/dev/node/src/main/resources/defaults.conf#L270) which would be rendered in the [POS contract](https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/genesis/contracts/ProofOfStake.scala#L10-L32)

- [x] Introduce POS multi-sig pub keys (pos-multi-sig-public-keys) in https://github.com/rchain/rchain/blob/dev/node/src/main/resources/defaults.conf#L250-L290
- [x] Pass these `pubKeys` configs in https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/genesis/contracts/ProofOfStake.scala#L10-L32
- [x] Introduce `posMultiSigQuorum` (pos-multi-sig-quorum) as a config parameter which defines how many confirmations are necessary to use multi-sig vault
  - this number should be checked to be less or equal the number of supplied public keys
- [x] Replace the hard-coded https://github.com/rchain/rchain/blob/888c8040892/casper/src/main/resources/PoS.rhox#L110-L112 pub keys and quorum data into parsing config with `$$` like https://github.com/rchain/rchain/blob/888c8040892/casper/src/main/resources/PoS.rhox#L240
- [ ] Add CLI options, it seems that rogach has support for lists e.g. --pub abc def.
  https://github.com/scallop/scallop/wiki/Option-types-and-definitions#trailing-arguments

### Verification

1. Start fresh node with 2 public keys for PoS multi-sig vault and quorum 1
2. Run exploratory deploy to get multi-sig REV address
    ```scala
    new return, lookup(`rho:registry:lookup`) in {
      new posCh, multiSigVaultCh in {
        lookup!(`rho:rchain:pos`, *posCh) |
        for(@(_, pos) <- posCh) {
          @pos!("getCoopVault", *multiSigVaultCh) |
          for(@(_, revAaddress, _) <- multiSigVaultCh) {
            return!(revAaddress)
          }
        }
      }
    }
    ```
3. Transfer 100 REVs to multi-sig REV address
    - needed to transfer REVs from multi-sig
5. Transfer 100 REVs to the first multi-sig public key (REV address)
    - needed to pay for deploy
6. Create deploy to transfer from multi-sig address to any other address and check that transfer is successful
    - variables are `${toAddr}` for target address (String) and `${amount}` (Number)
    ```scala
      new out(\`rho:io:stdout\`)
        , deployId(\`rho:rchain:deployId\`), deployerId(\`rho:rchain:deployerId\`)
        , lookup(\`rho:registry:lookup\`)
        , posCh, msRevVaultCh, revVaultCh, posMultiSigVaultCh, deployerKeyCh, toRevVaultCh, transferRetCh
      in {
        lookup!(\`rho:rchain:pos\`, *posCh) |
        lookup!(\`rho:rchain:multiSigRevVault\`, *msRevVaultCh) |
        lookup!(\`rho:rchain:revVault\`, *revVaultCh) |
        for(@(_, Pos) <- posCh ; @(_, MultiSigVault) <- msRevVaultCh ; @(_, RevVault) <- revVaultCh) {
          @Pos!("getCoopVault", *posMultiSigVaultCh) |
          @MultiSigVault!("deployerAuthKey", *deployerId, *deployerKeyCh) |
          @RevVault!("findOrCreate", "${toAddr}", *toRevVaultCh) |
          for(@(msRevVault, msRevAaddress, _) <- posMultiSigVaultCh ; _ <- toRevVaultCh ; @deployerKey <- deployerKeyCh) {
            @msRevVault!("transfer", "${toAddr}", ${amount}, deployerKey, *transferRetCh) |
            for(@transferRet <- transferRetCh) {
              deployId!(("Transfer result", transferRet))
            }
          }
        }
      }
    ```

### PR DESCRIPTION
**Pull request by stanislavlyalin**
## Overview

The PoS multi-sig public keys and PoS multi-sig quorum which was hard-coded in [PoS.rhox](https://github.com/rchain/rchain/blob/3371c8a400f4b82a7af833f25b9a0b6fc71709db/casper/src/main/resources/PoS.rhox) now can be configured in [defaults.conf](https://github.com/rchain/rchain/blob/4b36505e22ab46ba401775840d6f23920dc85dfa/node/src/main/resources/defaults.conf). Added runtime checking the `pos-multi-sig-quorum` value less or equal length of `pos-multi-sig-public-keys`.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: node/src/main/resources/defaults.conf [hunk 1]
```diff
@@ -287,6 +287,13 @@ casper {
 
     # genesis-block-number is used for hard fork on existing network.
     genesis-block-number = 0
+
+    # Public keys accepted by the PoS contract in the genesis block.
+    public-keys = [
```
Comment (tgrospic) [inline node/src/main/resources/defaults.conf]: Please rename this parameter to `pos-multi-sig-public-keys`.

Comment (stanislavlyalin) [reply node/src/main/resources/defaults.conf]: [Done](https://github.com/rchain/rchain/pull/3638/commits/175df4e90a472390fd321bad80790ef7303b479b).

File: node/src/main/resources/defaults.conf [hunk 2]
```diff
@@ -287,6 +287,17 @@ casper {
 
     # genesis-block-number is used for hard fork on existing network.
     genesis-block-number = 0
+
+    # Public keys accepted by the PoS contract in the genesis block.
+    pos-multi-sig-public-keys = [
+      04db91a53a2b72fcdcb201031772da86edad1e4979eb6742928d27731b1771e0bc40c9e9c9fa6554bdec041a87cee423d6f2e09e9dfb408b78e85a4aa611aad20c,
+      042a736b30fffcc7d5a58bb9416f7e46180818c82b15542d0a7819d1a437aa7f4b6940c50db73a67bfc5f5ec5b5fa555d24ef8339b03edaa09c096de4ded6eae14,
+      047f0f0f5bbe1d6d1a8dac4d88a3957851940f39a57cd89d55fe25b536ab67e6d76fd3f365c83e5bfe11fe7117e549b1ae3dd39bfc867d1c725a4177692c4e7754
+    ]
```
Comment (tgrospic) [inline node/src/main/resources/defaults.conf:296]: We will need better solution for defaults. Maybe with empty list and warning in the log. Let's keep it for now to speed up RPC release.

**Review by tgrospic — APPROVED** [review]
Nice work!

**Review by zsluedem — APPROVED** [review]
LGTM

**Comment by stanislavlyalin** [comment]
bors merge

**Comment by stanislavlyalin** [comment]
bors merge

**Pull request merged**